### ISSUE TITLE
Discussion: creating a Linked Data repository

### ISSUE THREAD

**Issue by pietercolpaert** [openaddresses/openaddresses#1098]
## Problem

In e.g., [Google Rich Snippets for events](https://developers.google.com/structured-data/rich-snippets/events) or in [DBPedia (wikipedia in Linked Data triples)](http://fragments.dbpedia.org/2014/en?subject=&predicate=http%3A%2F%2Fdbpedia.org%2Fontology%2Faddress&object=), addresses are indicated using a string, or with a "[blank node](https://en.wikipedia.org/wiki/Blank_node)" defined by a couple of hardly standardized strings.

This makes it hard for search engines, map applications, route planning systems, and so forth, to map descriptions of locations to WGS84 coordinates.

## Opportunity

What if we could link to an address using a URI? This would result in having global identifiers (the URIs) for each address in the world, which [dereferences](https://en.wikipedia.org/wiki/Dereferenceable_Uniform_Resource_Identifier) into different representation (e.g., HTML with a map indicating the address, with a WKT string, with a GeoJSON representation, with a JSON-LD representation, etc).

This way, the rich snippet would look something like this:
```json
{
  "@context": "http://schema.org",
  "@type": "Event",
  "name": "Example Band goes to San Francisco",
  "startDate" : "2013-09-14T21:30",
  "url" : "http://example.com/tourdates.html",
  "location" : {
    "@type" : "Place",
    "@id" : "http://id.openaddresses.io/123"
  }
}
```
And we would be able to _query_ dbpedia for things at a certain address

## What do we need

The only thing needed is that we have a tiny layer of software at http://id.openaddresses.io/{local_id} which does a 303 to a representation of that address using content-negotiation (e.g., a long lat pair using WKT, GeoJSON, JSON-LD, etc).

**Comment by iandees** [comment]
At first glance, this feels like it goes way beyond the scope of this project. Before we can expose any persistent IDs/URIs for addresses, we would be to come up with GUIDs for addresses and then merge duplicates/overlapping data into a single "canonical address" with that GUID. That's a pretty big project by itself and I don't think we're ready to figure it out, yet.

**Comment by migurski** [comment]
Yeah, we don’t have any concept of unique IDs and there are lots of duplicates. An `id` field in the conform could be a useful part of [the parcel discussion](https://github.com/openaddresses/machine/issues/163) though.

**Comment by pietercolpaert** [comment]
Indeed not a trivial task and interesting fact that it's beyond the scope of this project.

Slightly different idea:

How about creating a unique URI per address entry from a certain source? Then it's up to "future work" to create _same as_ links between IDs from different sources. A URI could then look something like this:
http://id.openaddresses.io/{datasourceid}/{localid}

Furthermore, we should only do this for datasets that have indicated that their IDs are persistent. This way, we will slowly start a world-wide discussion that we need persistent IDs for addresses?

**Comment by paulhoule** [comment]
If you want to "merge" duplicates the simple way to do is to have a statement that works like a permanent redirect (i.e. if you see X,  rewrite to Y) so that ultimately there is a "canonical form" of the address.  This is similar to owl:sameAs in spirit,  but it is better for a whole bunch of reasons,  least of which is that you get the right answers for SPARQL queries.

**Comment by iandees** [comment]
We have no interest (at least for now) in determining what the canonical address is. That's what I mean by "merging duplicates". There's still thousands of hours of work left in finding address data, which is what I'm interested in doing. If there's interest in taking this data we're finding and building some kind of normalization process on top of it, I would be extremely happy to see that happen.

**Comment by paulhoule** [comment]
So it is just statements of the form

{bag of address attributes} -> {enriched attributes such as WGS84}

that are in the system?

**Comment by sbma44** [comment]
I'm afraid that I'm also skeptical of adding a Linked Data aspect to the project. The ratio of time spent architecting such solutions to the number of actually implemented and useful projects is horrifically skewed. And as @iandees notes, there is still a ton of work to be done within this project's existing mission.

I hate to direct energy away from OpenAddresses, but the project that is closest to what you're considering is the burgeoning OpenStreetMap/Wikidata integration. There's plenty of work to be done there as well, but there already exist coherent namespaces and better means of resolving ambiguity. Advocacy within the OSM community could make it possible for OpenAddresses to play a role as well.

**Comment by migurski** [comment]
@paulhoule: for some value of “just”, yes. :smiley_cat: The authority model we are using here relies on the government or other address granter to provide a concept of uniqueness and identity. This seems to be best developed in the parcel use case, since [APN’s](https://en.wikipedia.org/wiki/Assessor%27s_parcel_number) are a well understood thing.

**Comment by pietercolpaert** [comment]
I'll close this issue now with this conclusion:

Open Addresses is an effort to create one dataset containing all address databases from over the entire world. For now, it has no ambition to remove "duplicates" to create a single "openaddresses id" per address. Therefore, it's unfeasible to create URI per address.

Furthermore, the local identifiers per address entry should not be converted to Linked Open Data as spending time on a low-quality Linked Data effort should not be a priority of this project: all time should be invested in harvesting more datasets.

The open addresses initiative however welcomes other initiatives to reuse their data, set up a persistent deduplication process, and to republish this data as Linked Open Data.
